// @sardine/eleventy-plugin-tinycss/dist/index.js
var $5hFZt$linkedom = require("linkedom");
var $5hFZt$purgecss = require("purgecss");
var $5hFZt$browserslist = require("browserslist");
var $5hFZt$parcelcss = require("@parcel/css");
var $5hFZt$fs = require("fs");

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}





async function $8d7f85d810401644$export$8e388a857ed0c062(rawCss, html, options) {
    try {
        var ref;
        const userPurgeCSSOptions = (ref = options === null || options === void 0 ? void 0 : options.purgeCSS) !== null && ref !== void 0 ? ref : {
        };
        var ref1;
        const targets = $5hFZt$parcelcss.browserslistToTargets(($parcel$interopDefault($5hFZt$browserslist))((ref1 = options === null || options === void 0 ? void 0 : options.browserslists) !== null && ref1 !== void 0 ? ref1 : null));
        const { code: code  } = $5hFZt$parcelcss.transform({
            filename: 'style.css',
            code: Buffer.from(rawCss),
            minify: true,
            targets: targets
        });
        const purgeCSSOptions = {
            content: [
                {
                    raw: html,
                    extension: 'html'
                }, 
            ],
            css: [
                {
                    raw: code.toString()
                }
            ]
        };
        Object.assign(purgeCSSOptions, userPurgeCSSOptions);
        const [{ css: css  }] = await new $5hFZt$purgecss.PurgeCSS().purge(purgeCSSOptions);
        return css;
    } catch (error) {
        console.error(error);
        throw error;
    }
}
function $8d7f85d810401644$export$eeda1f886239af9(link, root = '_site') {
    const src = root + link.href;
    return $5hFZt$fs.promises.readFile(src, {
        encoding: 'utf-8'
    });
}


const $2fed850edbc21d1f$export$45dcf71843d1d76c = async (html, options)=>{
    const { document: document  } = $5hFZt$linkedom.parseHTML(html);
    const styles = [
        ...document.querySelectorAll('style')
    ];
    const links = [
        ...document.querySelectorAll('link[rel=stylesheet]')
    ];
    if (links.length > 0 || styles.length > 0) {
        let css = '';
        if (links.length > 0) {
            const externalCSS = await Promise.all(links.map((link)=>{
                if (/^(https?\:\/\/|\/\/)/i.test(link.href)) return;
                return $8d7f85d810401644$export$eeda1f886239af9(link, options === null || options === void 0 ? void 0 : options.output);
            }));
            css = externalCSS.join('');
            links.map((link)=>{
                if (/^(https?\:\/\/|\/\/)/i.test(link.href)) return;
                return link.remove();
            });
        }
        if (styles.length > 0) for (const style of styles){
            css = css + style.textContent;
            style.remove();
        }
        const minicss = await $8d7f85d810401644$export$8e388a857ed0c062(css, html, options);
        const inline = document.createElement('style');
        const head = document.getElementsByTagName('head')[0];
        const inlinStyle = document.createTextNode(minicss);
        inline.appendChild(inlinStyle);
        head.appendChild(inline);
        html = document.toString();
    }
    return html;
};


module.exports = (eleventyConfig, options)=>{
    eleventyConfig.namespace('tinyCSS', ()=>{
        eleventyConfig.addTransform('tinyCSS', async (content, outputPath)=>{
            try {
                if (outputPath && outputPath.endsWith('.html')) content = await $2fed850edbc21d1f$export$45dcf71843d1d76c(content, options);
            } catch (error) {
                console.error(error);
            }
            return content;
        });
    });
};
